datetime/src/test/java/com/fasterxml/jackson/datatype/jsr310/ser/ZonedDateTimeSerTest.java

```diff
@@ -270,6 +273,37 @@ public void testSerializationAsStringWithZoneIdOn() throws Exception {
         assertEquals("The value is incorrect.", "\"" + DateTimeFormatter.ISO_ZONED_DATE_TIME.format(date) + "\"", value);
     }
 
+
+    @Test
+    public void testSerializationAsStringWithDefaultTimeZoneAndContextTimeZoneOn() throws Exception {
+		ZonedDateTime date = ZonedDateTime.now(Z3);
+		// With a custom DateTimeFormatter without a ZoneId.
+		String value = MAPPER.registerModule(new SimpleModule().addSerializer(new ZonedDateTimeSerializer(FORMATTER_WITHOUT_ZONEID))).writer()
+		        .with(TimeZone.getTimeZone(Z2))
```

Comment:
Any other tests needed for the various `SerializationFeature` combinations? Speaking of eclipse, it would be a useful exercise to run a code coverage (JaCoCo or similar) to see if any paths are missed.

Reply:
Taking a 2nd look, I think you have enough test coverage for `ZonedDateTimeSerTest`, given it really was a change to the if-check, so the only place I could see a few tests would be `OffsetDateTimeSerTest`.

Reply:
+1 for similar teest for `OffsetTimeSerTest`. Was wondering about whether config really should be through `SerializationFeature` as it only applies to this module (and since I hope to address "DateTimeConfig" for 2.14).
But given timing I guess let's go with `SerializationFeature` even if not optimal.

@vmeunier if you have a chance to add this test, I'm happy to merge the PR and can proceed with 2.13.0-rc1 very soon.

Reply:
@kupci @cowtowncoder 
Just so you know, I can't add a test with a custom formatter for OffsetDateTime since every constructor is protected.

You said you didn't want to add a public one (such as the one in ZonedDateTimeSerializer). So i'm not changing anything retarding this, but I still added some tests (in both ZonedDateTimeSerTest and OffsetDateTimeSerTest).

Reply:
Should I squash my commits ?
Do you have the autosquash ON on PR ?

Reply:
I can squash them, no worries.

As to custom formatter, I think custom  _format_ (via `@JsonFormat` and/or config overrides) is good to test -- but no need to manually construct instances with formatters. So only things users do via public API (including annotations), unless there is specific case reported that we think is good to cover.